### PR TITLE
Feature/sergeyro/icon button tooltipProps

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/3532714909/pulses/3716374138
`IconButton` new prop `tooltipProps` + `DialogPosition` refactoring
Better review commit by commit

image

### CODE REVIEW

File: src/components/IconButton/IconButton.tsx [hunk 1]
```diff
@@ -42,13 +43,16 @@ export interface IconButtonProps extends VibeComponentProps {
...
    */
   size?: Size;
-
   /**
    * Whether the tooltip should be displayed or not
    */
   hideTooltip?: boolean;
   /**
-   * Tooltip wraps the button icon, it will display in the tooltip, if not present the aria label will be shown
+   * Props for Tooltip wrapper component
```
Comment (hadasfa) [inline src/components/IconButton/IconButton.tsx:51]:
```suggestion
   * Props for Tooltip component
```

wrapper is redundant

File: src/components/IconButton/IconButton.tsx [hunk 2]
```diff
@@ -42,13 +43,16 @@ export interface IconButtonProps extends VibeComponentProps {
...
+   */
+  tooltipProps?: Partial<TooltipProps>;
+  /**
+   * Backward compatibility for props naming
```
Comment (hadasfa) [inline src/components/IconButton/IconButton.tsx:55]: why backward compatibility? in 99% of the cases developers will prefer to use this and not tooltip props. let's keep it and return the old comment.

File: src/helpers/backwardCompatibilityForProperties.ts [hunk 3]
```diff
@@ -1,8 +1,5 @@
 // eslint-disable-next-line default-param-last
-export function backwardCompatibilityForProperties(
-  valuesArrayByMostUpdateNaming: Array<string | boolean> = [],
-  defaultValue?: string | boolean
-) {
+export function backwardCompatibilityForProperties<T>(valuesArrayByMostUpdateNaming: Array<T> = [], defaultValue?: T) {
```
Comment (hadasfa) [inline src/helpers/backwardCompatibilityForProperties.ts:2]: awesome!

File: src/constants/positions.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ export const BASE_POSITIONS = {
   STRETCH: "Stretch"
 };
 
-export const DialogPosition = {
+export const DialogPositions = {
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply src/constants/positions.ts]: Converted it to enum - that's much better indeed

[the remark this answers is not in the document]
Comment (hadasfa) [reply src/constants/positions.ts]: Awesome!

File: src/components/IconButton/IconButton.tsx (hunk 2 above)
Comment (hadasfa) [reply src/components/IconButton/IconButton.tsx]: why this is still Backward compatibility for props naming ?  let's switch to old comment

File: src/components/IconButton/IconButton.tsx [hunk 5]
```diff
@@ -83,6 +87,8 @@ const IconButton: VibeComponent<IconButtonProps> & {
       id,
       icon,
       size,
+      tooltipProps,
+      // Backward compatibility for props naming
```
Comment (hadasfa) [inline src/components/IconButton/IconButton.tsx:90]: why backward backwardCompatibilityForProperties?

**Pull request merged**